services/wfm/src/main/java/org/openkilda/wfm/topology/stats/bolts/CacheBolt.java

```diff
@@ -116,6 +109,33 @@ private void initFlowCache(FlowRepository flowRepository) {
         }
     }
 
+    private Stream<FlowPath> extractAllFlowPaths(Flow flow) {
```

Comment:
This could be simpler:
```
return Stream.of(flow.getForwardPath(), flow.getReversePath(), flow.getProtectedForwardPath(), 
flow.getProtectedReversePath()).filter(Objects::nonNull).peek(p -> {
                    p.setSrcSwitch(flow.getSrcSwitch());
                    p.setDestSwitch(flow.getDestSwitch());
                })
```

Reply:
Fixed